### DISCUSSION TITLE
awesome bar - no keyboard shortcuts

### DISCUSSION THREAD

**Discussion by krackout** [bakkeby/dwm-flexipatch#227] [Q&A] [answered]
Apparently the awesome bar you've included is not the [latest by ornx](https://dwm.suckless.org/patches/awesomebar/dwm-awesomebar-20200907-6.2.diff). I cannot find the ability to add keyboard shortcuts to flexipatch's implementation.

Is it possible to be upgraded? Thank you.

**Comment by bakkeby** [comment, marked as answer]
The awesomebar patch adds more than one functionality:
- it provides evenly spaced window titles in the bar and
- it adds the possibility for clients to be hidden and
- it allows for certain actions happening when using mouse button clicks on the bar (as well as keybindings to hide windows)

This didn't fit very well in dwm-flexipatch as the click functionality and hidden support could easily be used for other window title patches such as fancybar or bartabgroups.

Because of this the functionality was separated and the awesomebar patch / toggle is reduced to simply adding evenly sized window titles.

The rest was moved to the wintitle actions toggle, which is enabled automatically when awesomebar is enabled. 

https://github.com/bakkeby/dwm-flexipatch/blob/4366f72390545fdc7ade62f4b002b5471913d0e5/patches.def.h#L406-L413

So you should look for this in your config.h to see what the keybindings are. 

https://github.com/bakkeby/dwm-flexipatch/blob/4366f72390545fdc7ade62f4b002b5471913d0e5/config.def.h#L968-L970

In general the keybindings listed on the suckless wiki page may be set up differently in dwm-flexipatch. This has to do with being able to try many combinations of patches without having to run into keybinding conflicts where one keybinding ends up doing more than one thing. Not every patch can use MOD+f.
